scripts/diff-coverage.py

```
@@ -0,0 +1,507 @@
+            try:
+                if int(counts.get(idx, 0)) > 0:
+                    covered.add(ln)
+            except (TypeError, ValueError):
+                continue
+        if measured:
+            out[_norm(path)] = (measured, covered)
+    return out
+
+
+def parse_cobertura(text: str) -> dict[str, tuple[set[int], set[int]]]:
+    """Cobertura XML（coverage.xml）→ <class filename>/<line number hits>"""
```

Ruff (0.16.1)

[warning] Docstring contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF002)